### ISSUE TITLE
[Android emulator] Flutter 3.13.0 failed host lookup

### ISSUE THREAD

**Issue by koodimetsa** [flutter/flutter#132840]
### Is there an existing issue for this?

- [X] I have searched the [existing issues](https://github.com/flutter/flutter/issues)
- [X] I have read the [guide to filing a bug](https://flutter.dev/docs/resources/bug-reports)

### Steps to reproduce

Upgraded from 3.10.6 -> 3.13.0

When doing a network call on Android emulator, a failed host lookup exception occurs on socket_patch.dart line 520. With iOS emulator this does not happen. Tested on two Android emulators, APIs 30 and 32.

On AndroidManifest internet permission is given `<uses-permission android:name="android.permission.INTERNET"/>`.

### Expected results

Network host lookup works.

### Actual results

Host loopup ends to exception.

### Code sample

Trying to add later.

### Flutter Doctor output

<details><summary>Doctor output</summary>

```console
[✓] Flutter (Channel stable, 3.13.0, on macOS 13.5 22G74 darwin-x64, locale en-FI)
[✓] Android toolchain - develop for Android devices (Android SDK version 30.0.3)
[✓] Xcode - develop for iOS and macOS (Xcode 14.3.1)
[✓] Chrome - develop for the web
[✓] Android Studio (version 2022.3)
[✓] VS Code (version 1.81.1)
[✓] Connected device (2 available)
[✓] Network resources

• No issues found!
```

</details>

**Comment by cyclotron3k** [comment]
I'm also seeing this issue after upgrading to Flutter 3.13.0. (Ubuntu 22.04, VSCode 1.81.1, testing on a Pixel 2 and a Pixel 6)

Seems to be very similar to these issues that were resolved in June:
https://github.com/flutter/flutter/issues/126650
https://github.com/dart-lang/sdk/issues/52487

As mentiond in one of those threads; disabling breakpoints for both "All exceptions" and "Uncaught Exceptions" is workaround.

**Comment by dam-ease** [comment]
Hi @koodimetsa. Thanks for filing this. Trying to reproduce on my end as well but does this workaround suit your case??

**Comment by koodimetsa** [comment]
@dam-ease Given workaround works 👍🏼

**Comment by zamirszn** [comment]
i upgraded to 3.13.0 im also having this issue

**Comment by koodimetsa** [comment]
@dam-ease This issue seems similar https://github.com/flutter/flutter/issues/132912

**Comment by MattL-NZ** [comment]
I'm facing a similar issue with upgrading to 3.13.0. Downgrading to 3.10.6 for now until it is resolved.

**Comment by hohlfma** [comment]
@dam-ease seems still not to be fixed, same issue here after upgrading from 3.10.6 --> 3.13.0.

Downgrading back to 3.10.6 solves the issue.

**Comment by koodimetsa** [comment]
I added a code sample which reproduces the exception with Test press 2.

**Comment by KostyaLocal** [comment]
Same error after upgrade to 3.13

**Comment by Tony0205** [comment]
Same Issue after upgrade to flutter 3.13

**Comment by dam-ease** [comment]
Hi everyone still facing this issue. Kindly checkout https://github.com/flutter/flutter/issues/126650 which has been closed as fixed and which I think the fix is on the `master` channel. Also kindly check this for more insights into a workaround https://github.com/flutter/flutter/issues/126650#issuecomment-1576503525

**Comment by koodimetsa** [comment]
There is now an open issue now at Dart repo: https://github.com/dart-lang/sdk/issues/53331 👍🏼

**Comment by KostyaLocal** [comment]
> Hi everyone still facing this issue. Kindly checkout #126650 which has been closed as fixed and which I think the fix is on the `master` channel. Also kindly check this for more insights into a workaround [#126650 (comment)](https://github.com/flutter/flutter/issues/126650#issuecomment-1576503525)
@dam-ease  Workaround is so-so. Stop ejecting all flutter exceptions including layout.

**Comment by mortenthansen** [comment]
Agree. The workaround is not good. This is triggered with only the "Uncaught Exception" flag enabled - what else are we ignoring without this?. I reverted to Flutter 3.10.6 until this is fixed. Not sure I understand how the issue can be closed.

**Comment by dam-ease** [comment]
Thanks for your response. This is being tracked per this https://github.com/dart-lang/sdk/issues/53334#issuecomment-1698645994

**Comment by jymden** [comment]
Having the same issue with Flutter 3.13.2

**Comment by omidi23** [comment]
Having the same issue with Flutter 3.13.2

**Comment by Wizzel1** [comment]
Same problem here at 3.13.2

**Comment by 4e6anenk0** [comment]
I had a similar issue on version 3.13 with a Linux program on a Linux host. Now, in version 3.14, this problem no longer occurs.

**Comment by cyclotron3k** [comment]
Seems to be fixed in 3.13.4 too :tada:

**Comment by yasinari** [comment]
The problem was solved with version 3.13.5.
